astropy/units/format/vounit.py

```diff
@@ -91,7 +92,12 @@ def parse(cls, s, debug=False):
             raise core.UnitsError(
                 "'{}' contains multiple slashes, which is "
                 "disallowed by the VOUnit standard".format(s))
-        result = cls._do_parse(s, debug=debug)
+        if len(s) > 2 and s[0] == s[-1] == '"':
+            cls._custom_inline = True
```

Comment:
This seems rather fragile, to change the class while parsing a given unit. I would suggest instead to overwrite `_do_parse` and call `_parse_unit` with an extra keyword argument. (And the move this logic inside there? Or could it even be inside `_parse_unit`?